src/ucai/core/databricks.py

```diff
@@ -111,6 +115,55 @@ def extract_function_name(sql_body: str) -> str:
     )
 
 
+def retry_on_session_expiration():
+    """
+    Decorator to retry a method upon session expiration errors with exponential backoff.
+
+    Args:
+        max_attempts (int): Maximum number of retry attempts.
+        base_delay (int): Base delay in seconds for exponential backoff.
+        max_delay (int): Maximum delay in seconds between retries.
+    """
+    max_attempts = int(UCAI_DATABRICKS_SESSION_RETRY_MAX_ATTEMPTS.get())
+
+    def decorator(func):
```

Comment:
Can we remove this and directly return wrapper? Update the function to retry_on_session_expiration(func).

Reply:
Yeah, I originally had the factory decorator for args passing of client-based overrides to retry logic. Forgot to remove this when moving to the ENV VAR approach. Cleaned up!